trie/receiptsTrie.go

```diff
@@ -0,0 +1,41 @@
+package trie
+
+import (
+	"fmt"
+
+	"github.com/multiversx/mx-chain-core-go/hashing"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-go/storage"
+)
+
+// GetLeafHashesAndPutNodesInRamStorage will return the leaf node hashes and put the rest of nodes in a storer
+func GetLeafHashesAndPutNodesInRamStorage(
+	branchNodesMap map[string][]byte,
+	db storage.Persister,
+	hasher hashing.Hasher,
+	marshaller marshal.Marshalizer,
+) ([][]byte, error) {
+	leafHashes := make([][]byte, 0)
+	for nodeHash, branchNodeSerialized := range branchNodesMap {
+		decodedNode, err := decodeNode(branchNodeSerialized, marshaller, hasher)
+		if err != nil {
+			return nil, err
+		}
+
+		hashes := decodedNode.getChildrenHashes()
+		if len(hashes) == 0 {
+			continue
+		}
+
+		leafHashes = append(leafHashes, hashes...)
```

Comment:
These are not only leaf hashes. An intermediary node can have another intermediary node as a child. So basically you collect here every hash except for the root hash.

Reply:
fixed.